src/Compilers/CSharp/Portable/Symbols/LocalSymbol.cs

```diff
@@ -25,6 +25,11 @@ protected LocalSymbol()
             get;
         }
 
+        /// <summary>
+        /// Syntax node that is used as the scope designator. Otherwise, null.
+        /// </summary>
+        internal abstract SyntaxNode ScopeDesignatorOpt { get; }
```

Comment:
would it be simpler to add another SynthesizedLocalKind for switch section locals?